winter/schema/type_inspection.py

```diff
@@ -179,6 +183,11 @@ def inspect_dataclass(hint_class) -> TypeInfo:
     return TypeInfo(type_=openapi.TYPE_OBJECT, properties=properties)
 
 
+@register_type_inspector(object, checker=lambda instance: getattr(instance, '__supertype__', None) is not None)
+def inspect_dataclass(hint_class) -> TypeInfo:
```

Comment:
`inspect_dataclass` -> `inspect_new_type` or something like this.

Reply:
Done